tools/diagram-sparql

```diff
@@ -142,7 +164,11 @@ while (1) {
 	# Strip leading whitespace and empty statements:
 	$all =~ s/\A\s*(\.\s*)*//sg;
       last if !$all;
-	#               1       2       34        5  6      7               8
+	$all =~ m/\A.+/;
+	my $firstLine = $&;
+	warn "PARSING all: $firstLine ...\n\n" if $debug;
+	# Parse into $s $p $v $terminator, and remove them from $all
+	#               1       2       34        5  6      7                8
```

Comment:
That is one ugly regex here.  The numbers in the comment seem to map to the regex groups.  But how to you get a tuple and terminator with 8 groups?  This is a bit confusing.